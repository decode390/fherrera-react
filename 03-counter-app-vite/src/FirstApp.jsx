import PropTypes from 'prop-types';

const newMessage = {
  message: 'Prueba1',
  title: 'Title'
}

const getName = ()=>'Nombre';

export const FirstApp = ({title, subTitle})=>{
  // if (!title) {
  //   throw new Error('Title no exits');
  // }

  return (
    <>
      {/*  <h1>{getName()}</h1>  */}
      <h1 data-testid="test-title">{title}</h1>
      {/*  <code>{ JSON.stringify(newMessage) }</code>  */}
      <p>{subTitle }</p>
      <p>{subTitle }</p>
    </>
  )
}

FirstApp.propTypes = {
  subTitle: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
  name: 'Miguel',
  subTitle: '123',
  // title: 'No title',
}
