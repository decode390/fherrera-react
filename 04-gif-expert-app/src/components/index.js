export * from './GifGrid';
export * from './AddCategory';
export * from './GiftItem';
