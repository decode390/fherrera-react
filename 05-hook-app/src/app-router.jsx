import {Navigate, createBrowserRouter} from 'react-router-dom';
import { MainApp, HomePage, LoginPage, AboutPage } from './09-useContext';
import ErrorPage from './error-page';

export const appRouter = createBrowserRouter([
  {
    path: "/",
    element: <MainApp />,
    children: [
      {
        path: "",
        element: <HomePage />,
      },
      {
        path: "login",
        element: <LoginPage />,
      },
      {
        path: "about",
        element: <AboutPage/>,
      },
      {
        path: "/*",
        element: <Navigate to="/login"/>,
      },

    ]
  },
]);
