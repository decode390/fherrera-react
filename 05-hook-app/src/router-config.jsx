import {Navigate, createBrowserRouter} from 'react-router-dom';
import { MainApp, HomePage, LoginPage, AboutPage } from './09-useContext';

export const routerConfig = [
  {
    path: "/",
    element: <MainApp />,
    children: [
      {
        path: "",
        element: <HomePage />,
      },
      {
        path: "login",
        element: <LoginPage />,
      },
      {
        path: "about",
        element: <AboutPage/>,
      },
      {
        path: "/*",
        element: <Navigate to="/login"/>,
      },

    ]
  },
];
