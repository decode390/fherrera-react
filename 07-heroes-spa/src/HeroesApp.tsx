import { AuthProvider } from "./auth/context"

export const HeroesApp = ({children}: any) => {
  return (
    <>
      <AuthProvider>
        <h1>HeroesApp</h1>
        { children }
      </AuthProvider>
    </>
  )
}
