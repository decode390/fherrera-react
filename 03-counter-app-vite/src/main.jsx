import React from 'react';
import ReactDOM from "react-dom/client";
import { HelloWorldApp } from './HelloWorldApp';
import { FirstApp } from './FirstApp';
import './index.css';
import { CounterApp } from './CounterApp';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <CounterApp value={12}/> 
    {/* <FirstApp title='Hello world'/> */}
  </React.StrictMode>
);

