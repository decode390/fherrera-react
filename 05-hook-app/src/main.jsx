import React from 'react'
import ReactDOM from 'react-dom/client'
import {
  createBrowserRouter,
  RouterProvider,
} from 'react-router-dom';
import './index.css'
import { routerConfig } from './router-config';

/* import "./08-useReducer/intro-reducer"; */

export const router = createBrowserRouter(routerConfig);
ReactDOM.createRoot(document.getElementById('root')).render(
  // <React.StrictMode>
    <RouterProvider router={router}/>
  // </React.StrictMode>,
)
