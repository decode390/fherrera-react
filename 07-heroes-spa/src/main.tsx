import './styles.css'
import 'animate.css';
import React from 'react'
import ReactDOM from 'react-dom/client'
import { RouterProvider, createBrowserRouter } from 'react-router-dom'
import { appRouter } from './router'
import { HeroesApp } from './HeroesApp';


const router = createBrowserRouter(appRouter);

ReactDOM.createRoot(document.getElementById('root')!).render(
  <React.StrictMode>
    <HeroesApp>
        <RouterProvider router={router}></RouterProvider>
    </HeroesApp>
  </React.StrictMode>,
)
