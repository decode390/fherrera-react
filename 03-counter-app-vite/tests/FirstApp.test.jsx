const { render } = require("@testing-library/react")
import {FirstApp} from "../src/FirstApp";


describe('Test in FirstApp', () => {
  test('should be match with snapshot', () => {
    const title = 'My title';
    const {container} = render(<FirstApp title={title}/>)

    expect(container).toMatchSnapshot();
  });

  test('should show title in h1', () => {
    const title = 'My title';
    const {container, getByText, getByTestId} = render(<FirstApp title={title}/>)

    expect(getByText(title)).toBeTruthy();
    const h1 = container.querySelector('h1');
    // expect(h1.innerHTML).toBe(title);
    // expect(h1.innerHTML).toContain(title);
    
    expect(getByTestId('test-title').innerHTML).toContain(title);
  });

  test('should show subtitle from props', () => {
    const title = 'My title';
    const subTitle = '123';
    const { getAllByText } = render(
      <FirstApp 
        title={title} 
        subTitle={subTitle}
      />
    )

    // expect(getByText(subTitle)).toBeTruthy();
    expect(getAllByText(subTitle).length).toBe(2);
  });
});
