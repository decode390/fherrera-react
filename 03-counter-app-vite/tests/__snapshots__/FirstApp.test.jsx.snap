// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in FirstApp should be match with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    My title
  </h1>
  <p>
    123
  </p>
  <p>
    123
  </p>
</div>
`;
